opentelemetry-sdk/src/logs/record.rs

```diff
@@ -201,6 +202,73 @@ impl SdkLogRecord {
             .flatten()
             .any(|(k, v)| k == key && v == value)
     }
+
+    /// Updates the first occurrence of an attribute with the specified key.
```

Comment:
Similar to `remove_attribute`, we should update all the occurrences of the given attribute key. For example, I think it would make it easier for redaction processor authors to redact a "secret" attribute with certainty.